packages/react/src/components/Theme/index.tsx

```diff
@@ -7,15 +7,22 @@
 
 import cx from 'classnames';
 import PropTypes from 'prop-types';
-import React, { useMemo } from 'react';
+import React, { useMemo, type PropsWithChildren } from 'react';
 import { usePrefix } from '../../internal/usePrefix';
 import { LayerContext } from '../Layer/LayerContext';
 
-export const ThemeContext = React.createContext({
+interface GlobalThemeProps {
+  theme?: 'white' | 'g10' | 'g90' | 'g100';
+}
```

Comment:
I have a question/suggestion. Can we also add a type `children?: React.ReactNode;` and get rid of the `GlobalTheme.propTypes`? Since the typescript will provides static type checking.
And if that make sense we could also approach the `Theme` with the same logic.

I'm curious about your thoughts? @mrkjdy @tay1orjones @andreancardona

Reply:
Would keeping the propTypes be useful for developers that are still using JavaScript? The react docs [recommend using TypeScript instead of using propTypes].

Reply:
Tha make sense. I just checked a few other .tsx files and it kept the `propTypes` also

Reply:
Yeah good question, we do need to keep the proptypes. It's mentioned in [the docs for adding component types](https://github.com/carbon-design-system/carbon/blob/main/docs/guides/adding-component-types.md#strategy).